tracecat/expressions/functions.py

```diff
@@ -353,11 +407,142 @@ def to_timestamp_str(x: datetime) -> float:
     return x.timestamp()
 
 
+def extract_second(x: datetime) -> int:
+    return x.second
+
+
+def extract_minute(x: datetime) -> int:
+    return x.minute
+
+
+def extract_hour(x: datetime) -> int:
+    return x.hour
+
+
+def extract_day_of_week(x: datetime, format: str = "number") -> int | str:
+    """Extract day of week from datetime. Returns 0-6 (Mon-Sun) or day name if format is "full" or "short"."""
+    weekday = x.weekday()
+    match format:
+        case "number":
+            return weekday
+        case "full":
+            days = [
+                "Monday",
+                "Tuesday",
+                "Wednesday",
+                "Thursday",
+                "Friday",
+                "Saturday",
+                "Sunday",
+            ]
+            return days[weekday]
+        case "short":
+            days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+            return days[weekday]
+        case _:
+            raise ValueError("format must be 'number', 'full', or 'short'")
+
+
+def extract_day(x: datetime) -> int:
+    return x.day
```

Comment:
can we rename this family of `extract_<unit>` to `get_<unit>` and add docstrings as well